lib/middleware/proxy-path.js

```diff
@@ -55,7 +55,7 @@ module.exports = function (rootUrl) {
         https.get({
             agent   : false,
             hostname: parsedUrl.hostname,
-            path    : url.resolve(parsedUrl.path, req.path),
+            path    : url.resolve(parsedUrl.path, pathWithIndex(req)),
```

Comment:
I think I'd rather see `req.path` get modified just before this `https.get()` call to contain "index.html" rather than calling out to `pathWithIndex()` here and in `onResponse()`.

Modifying it will ensure that any subsequent user of `req.path` will get the correct path without needing to explicitly call `pathWithIndex()` (which will be easy to forget to do).